native/java/org/jpype/classloader/JPypeClassLoader.java

```diff
@@ -0,0 +1,130 @@
+/*
+ * Copyright 2018, Karl Nelson
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+package org.jpype.classloader;
+
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.util.Map;
+import java.util.TreeMap;
+import java.util.jar.JarEntry;
+import java.util.jar.JarInputStream;
+
+/**
+ * Specialized class loader for jpype resources.
+ * <p>
+ * Loader to convert the internally stored resources into java classes. This
+ * prevents class load order problems when there are class dependencies.
+ * <p>
+ */
+public class JPypeClassLoader extends ClassLoader
+{
+  TreeMap<String, byte[]> map = new TreeMap<>();
+  static private JPypeClassLoader instance;
+
+  public static JPypeClassLoader getInstance()
+  {
+    if (instance == null)
+    {
+      JPypeClassLoader.instance = new JPypeClassLoader(getSystemClassLoader());
+    }
+    return instance;
+  }
+
+  private JPypeClassLoader(ClassLoader parent)
+  {
+    super(parent);
+  }
+
+  public void importClass(String name, byte[] code)
+  {
+    name = name.substring(0, name.length() - 6).replaceAll("/", ".");
```

Comment:
what happens here, if the the class is not found? Or name.length is < 6?

Reply:
Fixed.